### PR TITLE
Chore: Replace all usages of ioutil

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does**:
Since ioutil is [deprecated](https://golang.org/doc/go1.16#ioutil) since Go 1.16, replace all usages of it with preferred equivalents. Nothing important, but might as well clean up :)

**Which issue(s) this PR fixes**:

**Checklist**
- [x] Tests updated
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: ring/lifecycler_test.go [hunk 1]
```diff
@@ -596,7 +595,7 @@ func TestTokensOnDisk(t *testing.T) {
 	require.NoError(t, services.StartAndAwaitRunning(context.Background(), r))
 	defer services.StopAndAwaitTerminated(context.Background(), r) //nolint:errcheck
 
-	tokenDir, err := ioutil.TempDir(os.TempDir(), "tokens_on_disk")
+	tokenDir, err := os.MkdirTemp("", "tokens_on_disk")
```
Comment (pstibrany) [inline ring/lifecycler_test.go:598]: can we use `t.TempDir()`? No need to do manual deletion via `defer` then.

File: ring/tokens_test.go [hunk 2]
```diff
@@ -60,7 +59,7 @@ func TestTokens_Equals(t *testing.T) {
 }
 
 func TestLoadTokensFromFile_ShouldGuaranteeSortedTokens(t *testing.T) {
-	tmpDir, err := ioutil.TempDir("", "test-tokens")
+	tmpDir, err := os.MkdirTemp("", "test-tokens")
```
Comment (pstibrany) [inline ring/tokens_test.go:62]: Can we also replace this with `f.TempDir()`?

**Review by pstibrany — APPROVED** [review]
LGTM, thanks.

File: ring/lifecycler_test.go (hunk 1 above)
Comment (aknuds1) [reply ring/lifecycler_test.go:598]: @pstibrany good point, didn't think of it myself :)

File: ring/tokens_test.go (hunk 2 above)
Comment (aknuds1) [reply ring/tokens_test.go:62]: @pstibrany surely.

**Pull request merged**